### ISSUE TITLE
Incorrect grouping

### ISSUE THREAD

**Issue by darthsebulba04** [dbuezas/lovelace-plotly-graph-card#268]
**Describe the bug**
It looks like not all values are unit grouped correctly. Have a case where a Fahrenheit sensor is on the humidity axis. 

https://community.home-assistant.io/t/plotly-interactive-graph-card/347746/313?u=darthsebulba04

**Screenshots**

![](https://community-assets.home-assistant.io/original/4X/c/4/b/c4b4d017222452e03cc33f84c45626c9f088a1f1.jpeg)

**yaml**
```yaml
- type: custom:plotly-graph
            entities:
              - entity: input_number.weather_forecast_low
                legendgroup: 1
                connectgaps: true
                name: Low
                line:
                  width: 3
                  color: rgba(59,117,175,1)
              - entity: input_number.weather_forecast_high
                legendgroup: 2
                connectgaps: true
                name: High
                line:
                  width: 3
                  color: rgba(239,134,54,1)
              - entity: sensor.relative_humidity
                legendgroup: 3
                connectedgaps: true
                name: Humidity
                fill: tozeroy
              - entity: sensor.temperature
                legendgroup: 4
                connectgaps: true
                name: Temperature
                line:
                  width: 3
            hours_to_show: 24
            refresh_interval: 3600
            layout:
              paper_bgcolor: transparent
              plot_bgcolor: transparent
            config:
              staticPlot: true
            defaults:
              entity:
                show_value: true
                line:
                  shape: spline

**Comment by dbuezas** [comment]
Does input_number.weather_forecast_high have a unit_of_measurement?

**Comment by dbuezas** [comment]
E.g. do you see an axis title if you put it alone in a plot?

**Comment by darthsebulba04** [comment]
Yes, it has a unit of °F and it does show the unit on a standalone graph.
<img width="939" alt="image" src="https://user-images.githubusercontent.com/31055115/230907716-67801271-7ad5-495c-9913-8fbcb353a110.png">

I just did some more testing and it looks like it has to do with the order in which I have them listed in `entities:` .  If I put the temperature sensors first, then the humidity plots fine on its own axis (temp, temp, temp, humidity).  The way I have it currently is temp, temp, humidity, and then temp.  That seems to overwrite the unit on the second Y axis.  Changing the `legendgroup` didn't make a difference.

**Comment by dbuezas** [comment]
I see, it does look like a bug in the way in which I assign units to axes

**Comment by darthsebulba04** [comment]
Makes sense, awesome.  Always glad to throw monkey wrenches in things ;)

**Comment by dbuezas** [comment]
Should be fixed in v3.3.3. Let me know if updating fixes it for you too ;)
